src/java.desktop/windows/native/libawt/windows/awt_Window.cpp

```diff
@@ -3211,18 +3261,36 @@ void AwtWindow::_UpdateWindow(void* param)
 {
     JNIEnv *env = (JNIEnv *)JNU_GetEnv(jvm, JNI_VERSION_1_2);
 
-    UpdateWindowStruct *uws = (UpdateWindowStruct *)param;
+    UpdateWindowStruct *uws = static_cast<UpdateWindowStruct *>(param);
     jobject self = uws->window;
     jintArray data = uws->data;
 
-    PDATA pData;
-    JNI_CHECK_PEER_GOTO(self, ret);
-    AwtWindow *window = (AwtWindow *)pData;
+    AwtWindow *window = nullptr;
+    if (self == NULL) {
+        env->ExceptionClear();
+        JNU_ThrowNullPointerException(env, "self");
+        if (data != NULL) {
+            env->DeleteGlobalRef(data);
+        }
+        delete uws;
+        return;
+    } else {
+        window = (AwtWindow *) JNI_GET_PDATA(self);
+        if (window == nullptr) {
+            THROW_NULL_PDATA_IF_NOT_DESTROYED(self);
+            env->DeleteGlobalRef(self);
+            if (data != NULL) {
+                env->DeleteGlobalRef(data);
+            }
+            delete uws;
+            return;
+        }
+    }
 
     window->UpdateWindow(env, data, (int)uws->width, (int)uws->height,
                          uws->hBitmap);
 
-  ret:
+
```

Comment:
I recommend deleting this line, since it does not make sense to have two consecutive blank lines here.